changes: add before-4.0 changes

bagder opened the pull request:
Archeology and researching CHANGES files fom old urlget and httpget tarballs made me able to recrate changelogs for all releases back to httpget 0.1. Some of them lack details. Some of them have gotten estimated release dates as not all dates were clearly noted.

As these changes are added in the proper release change+bugfix syntax, they will bump the numbers in the release log and elsewhere accordingly. These are **29** releases done before the rename to curl.

dfandrich commented:
You could change the script so the "origin" is ver. 4.0, making previous
releases get a negative release number so not breaking future release
numbering. This can be justified since the previous releases weren't
technically "curl".

bagder commented:
I could, but does it matter much if we just "rebase" the count to start from httpget 0.1 instead of curl 4.0?

In some aspects it is the same thing that continued, it was just the name that changed...

dfandrich commented:
Mostly the release announcements that say "Public curl releases: X" and things
like videos saying "Release 221" will suddenly be inconsistent.

bagder commented:
> inconsistent

True, but A) it looks very odd to count to minus and B) we've corrected the release numbering in the past when we've adjusted the (past) counting...

bagder commented:
This how the bottom of the table looks like if we want to maintain the previous release number at the top:

![Screenshot 2023-08-14 at 08-17-44 curl - Release Table](https://github.com/curl/curl-www/assets/177011/47bde28d-abd1-4f45-baac-00f3505513db)

rybak commented:
Nice! A couple of typos in the commit message:

> Archeology and researching CHANGES files fom old urlget and httpget

s/fom/from/

> tarballs made me able to recrate changelogs for all releases back to

s/recrate/recreate/

> httpget 0.1. Some of them lack details. Some of them have gotten
> estimated release dates as not all dates were clearly noted.

bagder commented:
Let's try and see how it looks...

dfandrich commented:
> True, but A) it looks very odd to count to minus and B) we've corrected the
> release numbering in the past when we've adjusted the (past) counting...

If you've renumbered it before, then I guess it's not that big a deal. Also,
future release numbers will get larger with a jump, not smaller, so there won't
be two videos or announcements where different releases are called the same
number.

rybak commented on _changes.html:
```diff
@@ -12442,8 +12442,354 @@
+   login with a weird email address as password).
+</ul>
+
+<hr>
```
Noticed on the [website](https://curl.se/changes.html) that separator `<hr>` is only used between 1.5 and 2.0 and between 3.12 and 4.0. The horizontal ruler is not present between:

- 0.3 and 1.0
- 2.9 and 3.0
- 4.10 and 5.0
- 5.11 and 6.0
- 6.5.2 and 7.1 (sidenote: where is 7.0?)

bagder replied:
I put the `<hr>` a little half-assed where there was a rename: when httpget become urlget and when urlget become curl. We could probably come up with something better to visualized that.

There was no 7.0 release. We did a series of beta releases and then we bumped to 7.1 to avoid any confusions. You can see that there are a few other gaps in the release train as well.
